hatasmota/camera.py

```diff
@@ -0,0 +1,131 @@
+"""Tasmota camera."""
+
+from __future__ import annotations
+
+from collections.abc import Awaitable
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from aiohttp import ClientResponse, ClientSession
+
+from .const import CONF_DEEP_SLEEP, CONF_IP, CONF_MAC
+from .entity import (
+    TasmotaAvailability,
+    TasmotaAvailabilityConfig,
+    TasmotaEntity,
+    TasmotaEntityConfig,
+)
+from .mqtt import ReceiveMessage
+from .utils import (
+    config_get_state_offline,
+    config_get_state_online,
+    get_topic_command,
+    get_topic_command_state,
+    get_topic_stat_result,
+    get_topic_tele_sensor,
+    get_topic_tele_state,
+    get_topic_tele_will,
+    get_value_by_path,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class TasmotaCameraConfig(TasmotaAvailabilityConfig, TasmotaEntityConfig):
+    """Tasmota camera configuation."""
+
+    idx: int
+
+    command_topic: str
+    result_topic: str
+    state_topic: str
+    sensor_topic: str
+    ip_address: str
+
+    @classmethod
+    def from_discovery_message(cls, config: dict, platform: str) -> TasmotaCameraConfig:
+        """Instantiate from discovery message."""
+        return cls(
+            endpoint="camera",
+            idx=0,
+            friendly_name=None,
+            mac=config[CONF_MAC],
+            platform=platform,
+            poll_payload="",
+            poll_topic=get_topic_command_state(config),
+            availability_topic=get_topic_tele_will(config),
+            availability_offline=config_get_state_offline(config),
+            availability_online=config_get_state_online(config),
+            deep_sleep_enabled=config[CONF_DEEP_SLEEP],
+            command_topic=get_topic_command(config),
+            result_topic=get_topic_stat_result(config),
+            state_topic=get_topic_tele_state(config),
+            sensor_topic=get_topic_tele_sensor(config),
+            ip_address=config[CONF_IP],
+        )
+
+
+class TasmotaCamera(TasmotaAvailability, TasmotaEntity):
+    """Representation of a Tasmota camera."""
+
+    _cfg: TasmotaCameraConfig
+
+    def __init__(self, **kwds: Any):
+        """Initialize."""
+        self._sub_state: dict | None = None
+        super().__init__(**kwds)
+
+    async def subscribe_topics(self) -> None:
+        """Subscribe to topics."""
+
+        def state_message_received(msg: ReceiveMessage) -> None:
```

Comment:
What is this used for? The core PR doesn't seem to use it?

Reply:
It was part of the original version of the core PR. It was removed during the review.
I have updated this PR to reflect that.